File: 165_Compare_Version_Numbers.cpp
class Solution {
private:
    int levelRevision(string& version, int& index) {
        int level = 0;
        while (index < version.size() && version[index] != '.')
            level = level * 10 + (version[index++] - '0');
        index++;
        return level;
    }
public:
    int compareVersion(string version1, string version2) {
        int i = 0, j = 0;
        while (i < version1.size() || j < version2.size()) {
            int a = levelRevision(version1, i), b = levelRevision(version2, j);
            if (a > b) return 1;
            else if (a < b) return -1;
        }
        return 0;
    }
};

File: 165_Compare_Version_Numbers.py
class Solution:
    def compareVersion(self, version1: str, version2: str) -> int:
        v1l, v2l = version1.split('.'), version2.split('.')
        for i in range(max(len(v1l), len(v2l))):
            v1 = int(v1l[i]) if i < len(v1l) else 0
            v2 = int(v2l[i]) if i < len(v2l) else 0
            if v1 != v2:
                return 1 if v1 > v2 else -1
        return 0
